src/text_render.rs

```diff
@@ -514,9 +516,9 @@ where
                 left: image.left,
             },
         );
-    }
-
-    let details = atlas.glyph(&cache_key).unwrap();
+        inner.glyphs_in_use.insert(cache_key);
+        inner.glyph_cache.get(&cache_key).unwrap()
```

Comment:
An alternative implementation for this block is:

```rs
        inner.glyphs_in_use.insert(cache_key);
        inner.glyph_cache.get_or_insert(cache_key, || GlyphDetails {
            width: image.width,
            height: image.height,
            gpu_cache,
            atlas_id,
            top: image.top,
            left: image.left,
        })
```

I'm unsure which you prefer. This avoids the `.unwrap()` but is likely a little less readable.

Reply:
I added this in https://github.com/grovesNL/glyphon/pull/121/commits/52c099ca954b4d5cdf711fa72e914077528b8e0f - happy to revert.